src/calendar.interface.ts

```diff
@@ -1,11 +1,24 @@
 import { EventEmitter, TemplateRef } from '@angular/core';
+import 'moment-recur-ts'
+import { Rule } from 'moment-recur-ts';
 
+export interface IOccurence {
+    eventUTCStartTime: number;
+    eventUTCEndTime: number;
+}
+
+export interface IOccurences {
+
+}
 export interface IEvent  {
     allDay: boolean;
     endTime: Date;
     startTime: Date;
     title: string;
     category?: string;
+    rruleFreq?: Rule.MeasureInput;
+    rruleInterval?: Rule.UnitsInput;
+    rruleUntil?: Date;
```

Comment:
Instead of introducing a separate rruleUntil field on IEvent, could we just leverage the existing endTime field? Basically startTime and endTime field determine the total period of recurring event, rruleFreq and rruleInterval determine what's the pattern of the recurring event.

Reply:
if endTime is needed, maybe I didn't get the logic in below code. eventUTCEndTime is only set based on eventUTCStartTime. 
eventUTCEndTime: eventUTCStartTime + 86400000


```
for( let match of moment(event.startTime.getDate()).recur(utcStartTime,utcEndTime).every(event.rruleInterval, event.rruleFreq).all('L') ){
            let matchDate = new Date(match)
            let eventUTCStartTime = Date.UTC(matchDate.getFullYear(), matchDate.getMonth(), matchDate.getDate())
            occurences.push({
                eventUTCStartTime: eventUTCStartTime,
                eventUTCEndTime: eventUTCStartTime + 86400000
            })
        }
```

Reply:
moment(date).recure(start,end) delivers the list of dates when an event recures.
So eventUTCStartTime is not the start time of the event but the start time of the day on which the event happens. That is why I've selected as eventUTCEndTime the end time of this day. May be adding 86399999 is correct. ( I'm a chemist not a mathematician :-)

Reply:
make sense